lib/travis/build/script/dart.rb

```diff
@@ -114,10 +114,11 @@ def script
         private
 
           def archive_url
-            if not ["stable", "dev"].include?(config[:dart])
-              sh.failure "Only 'stable' and 'dev' can be used as dart version for now"
+            if ["stable", "dev"].include?(config[:dart])
+              "https://storage.googleapis.com/dart-archive/channels/#{config[:dart]}/release/latest"
+            else 
+              "https://storage.googleapis.com/dart-archive/channels/#{config[:dart]}"
```

Comment:
IIUC the user will have to use `stable/release/1.14.2` or `dev/release/1.16.0-dev.1.0` ?

According to https://www.dartlang.org/downloads/archive/ it looks like all stable versions are also available on dev channel. If it is intended we could have :

```ruby
# support of "dev" or "start"
if ["stable", "dev"].include?(config[:dart])
  "https://storage.googleapis.com/dart-archive/channels/#{config[:dart]}/release/latest"
# support of "stable/release/1.15.0" or "be/raw/110749"
elsif config[:dart].include?("/")
  "https://storage.googleapis.com/dart-archive/channels/#{config[:dart]}"
# support of "1.15.0" or "1.16.0-dev.2.0"
else
  "https://storage.googleapis.com/dart-archive/channels/dev/release/#{config[:dart]}"
```

WDYT ?